components/LaunchPad/Token/CreateToken.tsx

```diff
@@ -90,22 +93,133 @@ export const CreateToken = () => {
+  }
+
+  const isValidAddr = async (addr: string) => {
+    //Check your if account exist on Algorand
+    try {
+      await algodClient().accountInformation(addr).do()
+      return true
+    } catch (error) {
+      return false
+    }
   }
 
   const onSubmit = async (e) => {
     e.preventDefault()
+    let _error = false
+    if (
+      !tokenName ||
+      !unitName ||
+      !totalSupply ||
+      !decimals ||
+      (showClawbackAddr && !clawbackAddr) ||
+      (showReserveAddr && !reserveAddr) ||
+      (showFreezeAddr && !freezeAddr) ||
+      (showManagerAddr && !managerAddr)
+    ) {
+      setError((prev) => ({ ...prev, all: 'Please enter all the required field' }))
+      _error = true
+    }
+
+    //Ensure it is a valid whole number
+    if (isNaN(totalSupply) || !isInteger(totalSupply) || totalSupply < 1) {
+      setError((prev) => ({ ...prev, totalSupply: 'Enter a valid whole number' }))
+
+      _error = true
+    }
+
+    //Ensure it is a valid whole number and between 1 - 10
+    if (isNaN(decimals) || !isInteger(decimals) || decimals < 1 || decimals > 10) {
+      setError((prev) => ({ ...prev, decimals: 'Enter whole number between 1 and 10' }))
+
+      _error = true
+    }
 
+    //Check for string with no spaces
+    if (/\s/g.test(tokenName.trim())) {
+      setError((prev) => ({ ...prev, tokenName: 'Spaces not allowed between letters!' }))
+      _error = true
+    }
+    if (/\s/g.test(unitName.trim())) {
+      setError((prev) => ({ ...prev, unitName: 'Spaces not allowed between letters!' }))
+      _error = true
+    }
+
+    //Check for valid url string
+    if (assetURL && !isValidURL(assetURL.trim())) {
+      setError((prev) => ({ ...prev, assetURL: 'Invalid URL string!' }))
+      _error = true
+    }
+
+    //Confirm its a valid algorand address
+    if (showClawbackAddr && clawbackAddr && !(await isValidAddr(clawbackAddr.trim()))) {
+      setError((prev) => ({ ...prev, clawbackAddr: 'Invalid Algorand address!' }))
+      _error = true
+    }
+
+    if (showReserveAddr && reserveAddr && !(await isValidAddr(reserveAddr.trim()))) {
+      setError((prev) => ({ ...prev, reserveAddr: 'Invalid Algorand address!' }))
+      _error = true
+    }
+
+    if (showFreezeAddr && freezeAddr && !(await isValidAddr(freezeAddr.trim()))) {
+      setError((prev) => ({ ...prev, freezeAddr: 'Invalid Algorand address!' }))
+      _error = true
+    }
+
+    if (showManagerAddr && managerAddr && !(await isValidAddr(managerAddr.trim()))) {
+      setError((prev) => ({ ...prev, managerAddr: 'Invalid Algorand address!' }))
+      _error = true
+    }
+
+    if (!_error) {
+      setError(null)
+      createToken()
+    }
+  }
+
+  const createToken = () => {
     setLoading(true)
+    const payload = { ...formData, decimals: Number(decimals), totalSupply: Number(totalSupply) }
+    delete payload.showClawbackAddr
+    delete payload.showFreezeAddr
+    delete payload.showManagerAddr
+    delete payload.showReserveAddr
```

Comment:
I do not understand why you are deleting these properties from the payload. Can you explain your thought process?

Reply:
These are not part of what is required to create a token, so I deleted them from the payload object. I am only using they for the UI toggle.

Reply:
Ahh I see my mistake, I misread it. I thought you were deleting the actual properties not the boolean used for the UI toggle.